pkg/vector/Dockerfile

```diff
@@ -1,12 +1,78 @@
 # Copyright (c) 2025 Zededa, Inc.
 # SPDX-License-Identifier: Apache-2.0
 
-FROM paulzededa/eve-vector:no-lua
+ARG RUST_VERSION=lfedge/eve-rust:1.85.1-2
+ARG VECTOR_FEATURES='--no-default-features --features \
+sources-socket,\
+sources-internal_metrics,\
+transforms-aws_ec2_metadata,\
+transforms-dedupe,\
+transforms-filter,\
+transforms-window,\
+transforms-log_to_metric,\
+transforms-metric_to_log,\
+transforms-reduce,\
+transforms-remap,\
+transforms-route,\
+transforms-exclusive-route,\
+transforms-sample,\
+transforms-throttle,\
+sinks-socket,\
+sources-prometheus-scrape,\
+sinks-prometheus'
 
-RUN apk add --no-cache inotify-tools=4.23.9.0-r0
+# hadolint ignore=DL3006
+FROM --platform=$BUILDPLATFORM ${RUST_VERSION} AS toolchain-base
+ARG TARGETARCH
+
+FROM toolchain-base AS target-amd64
+ENV CARGO_BUILD_TARGET="x86_64-unknown-linux-musl"
+
+FROM toolchain-base AS target-arm64
+ENV CARGO_BUILD_TARGET="aarch64-unknown-linux-musl"
+
+FROM toolchain-base AS target-riscv64
+ENV CARGO_BUILD_TARGET="riscv64gc-unknown-linux-gnu"
```

Comment:
I would prefer to remove RISC-V "support" from the project, but for now, are you sure GNU works here? I don't care if it does not, though...